transmart-core-db/grails-app/services/org/transmartproject/db/userqueries/UserQueryService.groovy

```diff
@@ -1,46 +1,192 @@
 import org.transmartproject.core.userquery.UserQueryResource
+import org.transmartproject.core.userquery.UserQuerySetResource
+import org.transmartproject.core.users.AuthorisationChecks
+import org.transmartproject.core.users.LegacyAuthorisationChecks
+import org.transmartproject.core.users.PatientDataAccessLevel
 import org.transmartproject.core.users.User
+import org.transmartproject.db.clinical.ConstraintAccessChecker
 import org.transmartproject.db.querytool.Query
 
+import java.util.stream.Collectors
+
 @Transactional
+@CompileStatic
 class UserQueryService implements UserQueryResource {
 
-    @Override
-    List<UserQuery> list(User currentUser) {
-        Query.findAllByUsernameAndDeleted(currentUser.username, false)
+    @Autowired
+    UserQuerySetResource userQuerySetResource
+
+    @Autowired
+    AuthorisationChecks authorisationChecks
+
+    @Autowired
+    LegacyAuthorisationChecks legacyAuthorisationChecks
+
+    @Autowired
+    ConceptsResource conceptsResource
+
+    @Autowired
+    SessionFactory sessionFactory
+
+
+    private static void validateSubscriptionEnabled(Boolean subscribed, SubscriptionFrequency subscriptionFreq) {
+        boolean subscriptionFreqSpecified = subscriptionFreq != null
+        boolean subscriptionEnabled = Holders.config.getProperty('org.transmartproject.notifications.enabled', Boolean)
+        if (!subscriptionEnabled && (subscribed || subscriptionFreqSpecified)) {
+            throw new ServiceNotAvailableException(
+                    "Subscription functionality is not enabled. Saving subscription data not supported.")
+        }
     }
 
-    @Override
-    List<UserQuery> listSubscribed() {
-        Query.findAllByDeletedAndSubscribed(false, true)
+    static UserQueryRepresentation toRepresentation(UserQuery query) {
+        query.with {
+            new UserQueryRepresentation(
+                    id,
+                    name,
+                    patientsQuery ? ConstraintFactory.read(patientsQuery) : null,
+                    observationsQuery ? BindingHelper.objectMapper.readValue(observationsQuery, Object) : null,
+                    apiVersion,
+                    bookmarked,
+                    subscribed,
+                    subscriptionFreq,
+                    queryBlob ? BindingHelper.objectMapper.readValue(queryBlob, Object) : null,
+                    createDate,
+                    updateDate,
+            )
+        }
+    }
+
+    void checkConstraintAccess(Constraint constraint, User currentUser) {
+        new ConstraintAccessChecker(currentUser, PatientDataAccessLevel.MEASUREMENTS,
+                authorisationChecks, legacyAuthorisationChecks, conceptsResource)
+                .build(constraint)
     }
 
     @Override
-    UserQuery get(Long id, User currentUser) {
-        Query query = Query.findByIdAndDeleted(id, false)
+    List<UserQueryRepresentation> list(User currentUser) {
+        def result = Query.createCriteria().list {
```

Comment:
So should we always use `createCriteria` over Gorm's `findBy`?

Reply:
We do if we use `@CompileStatic`. This will change in Grails 3.3, which provides `@GrailsCompileStatic` which is aware of domain classes. Note that the argument of `list` is a closure, which is not compiled statically. (So for full static compilation, this is not sufficient).

Reply:
ok, thanks for the explanation, I will remember this next time